### ISSUE TITLE
query_sources fails if network is not accessible

### ISSUE THREAD

**Issue by mbjones** [cboettig/contentid#81]
One advantage of `contentid` is the ability to disconnect from the network and still operate on resources that contentid has cached in the local registry. However, even if a resource has been cached locally, some operations produce errors when the network is not available. For example, here's the error produced without the network even when an object was cached locally:

```r
> delta_catch_id
[1] "hash://sha256/e0dc10d7f36cfc5ac147956abb91f24f2b2df9f914a004bbf1e85e7d9cf52f41"
> contentid::query_sources(delta_catch_id)
                                                                                                                                                source                          
1   /Users/jones/Library/Application Support/org.R-project.R/R/contentid/sha256/e0/dc/e0dc10d7f36cfc5ac147956abb91f24f2b2df9f914a004bbf1e85e7d9cf52f41
2 https://gmn.edirepository.org/mn/v2/object/https%3A%2F%2Fpasta.lternet.edu%2Fpackage%2Fdata%2Feml%2Fedi%2F233%2F2%2F015e494911cf35c90089ced5a3127334
                 date
1 2021-10-30 16:11:35
2 2020-04-12 15:05:21
> # disconnect from Internet here
> contentid::query_sources(delta_catch_id)
Error in rbind(deparse.level, ...) :                                                                                                                                            
  numbers of columns of arguments do not match
In addition: Warning messages:
1: Error in curl::curl_fetch_memory(file, handle): LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 60
 
2: In curl::curl_fetch_memory(url, handle = handle) :
  Could not resolve host: archive.softwareheritage.org
3: In curl::curl_fetch_memory(url, handle = handle) :
  Could not resolve host: cn.dataone.org
4: In curl::curl_fetch_memory(url, handle = handle) :
  Could not resolve host: zenodo.org
> 
```

These should be resolvable by using `tryCatch` and return any results that are possible, and just print a warning if some registries that were requested are not accessible.

**Comment by cboettig** [comment]
Definitely a bug.  Looks like the URL calls are timing out without throwing any errors, note that `In addition: Warning messages:` appears before all of the `curl` errors, indicating these are already intercepted by `tryCatch`.  Unfortunately, it also looks like the tryCatch return object is not in the correct schema, so the function fails to combine the 0-row object with the results found from the local registry.  

However, note that `resolve()` should still work, and work fast -- it shouldn't even wait for timeout.  `query_sources()` is designed to show all sources, `resolve()` only cares about finding one.  

You can also set `all = FALSE` as an option:

```
query_sources(delta_catch_id, all = FALSE)
```

which should work and work fast as well (this is what resolve uses internally).  Note that `all=FALSE` will never display remote sources if a local source is found, because it doesn't even touch the internet.  

I think for most end-user use cases, users probably want `resolve()` and not `query_sources()`?  (Also, in the latest github version, `query_sources()` has been renamed `sources()` )

We should fix the bug, but `query_sources()` in offline mode with default registries will still be slow when `all=TRUE`, since it will have to wait for each one to timeout.

**Comment by cboettig** [comment]
(see #74)

**Comment by cboettig** [comment]
should be fixed by #88 (?)
